pinot-core/src/main/java/org/apache/pinot/core/plan/AggregationPlanNode.java

```diff
@@ -62,31 +68,164 @@ public AggregationPlanNode(IndexSegment indexSegment, QueryContext queryContext)
   public Operator<IntermediateResultsBlock> run() {
     assert _queryContext.getAggregationFunctions() != null;
 
+    boolean hasFilteredPredicates = _queryContext.isHasFilteredAggregations();
+    if (hasFilteredPredicates) {
+      return buildFilteredAggOperator();
+    }
+
+    return buildNonFilteredAggOperator();
```

Comment:
(minor) Can be simplified for better readability
```suggestion
    return _queryContext.isHasFilteredAggregations() ? buildFilteredAggOperator() : buildNonFilteredAggOperator();
```

Reply:
Fixed